### PR TITLE
Implement launch single task execution

### PR DESCRIPTION
**Pull request by katrogan**
# TL;DR
This allows users to execute **a previously registered** task individually. Next up is register_and_execute for notebook and other usage.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
This change does some refactoring to clarify the distinction between **launch** and **execute** such that launch now refers to an entity which can be used to _trigger_ an execution, meanwhile execute still refers to something that can be individually executed.

## Tracking Issue
https://github.com/lyft/flyte/issues/256

## Follow-up issue
https://github.com/lyft/flyte/issues/221

### CODE REVIEW

**Comment by wild-endeavor** [comment]
Can you give some insight into what get_node is used for?  That's the bit that's confusing to me.

File: flytekit/common/nodes.py [hunk 1]
```diff
@@ -100,7 +103,8 @@ def _return_mapping_object(self, sdk_node, sdk_type, name):
         return _promise.NodeOutput(sdk_node, sdk_type, name)
 
 
-class SdkNode(_six.with_metaclass(_sdk_bases.ExtendedSdkType, _hash_mixin.HashOnReferenceMixin, _workflow_model.Node)):
+class SdkNode(_six.with_metaclass(_sdk_bases.ExtendedSdkType, _hash_mixin.HashOnReferenceMixin, _workflow_model.Node,
```
[the remark this answers is not in the document]
Comment (katrogan) [reply flytekit/common/nodes.py]: updated

**Comment by katrogan** [comment]
> Can you give some insight into what get_node is used for?

that was stale, thanks for the catch. should be cleaned up

File: flytekit/common/tasks/task.py [hunk 2]
```diff
@@ -252,3 +255,28 @@ def __repr__(self):
             task_type=self.type,
             interface=self.interface
         )
+
+    @_exception_scopes.system_entry_point
+    def launch(self, **input_map):
```
[the remark this answers is not in the document]
Comment (katrogan) [reply flytekit/common/tasks/task.py:260]: it defines an execute method with a different signature than what the task already implements. i could refactor the task's execute method but to be fair it also does handle execution

**Comment by katrogan** [comment]
friendly ping @wild-endeavor @EngHabu

**Comment by katrogan** [comment]
@EngHabu mind taking another look? I refactored to use Launch in place of Execute like we discussed (and deprecated existing calls to launch_foo) however I wasn't sure if it made sense to change the flyte-cli command too along the same lines?

File: flytekit/common/mixins/launchable.py [hunk 3]
```diff
@@ -35,13 +34,28 @@ def execute(self, project, domain, inputs=None, name=None, notification_override
             annotation_overrides=annotation_overrides,
         )
 
+    def execute(self, project, domain, inputs=None, name=None, notification_overrides=None, label_overrides=None,
+                annotation_overrides=None):
+        """
+        Deprecated. Use launch instead.
+        """
+        return self.launch_with_literals(
```
[the remark this answers is not in the document]
Comment (katrogan) [reply flytekit/common/mixins/launchable.py]: thanks for the catch, done

File: flytekit/common/mixins/launchable.py [hunk 4]
```diff
@@ -35,13 +34,28 @@ def execute(self, project, domain, inputs=None, name=None, notification_override
             annotation_overrides=annotation_overrides,
         )
 
+    def execute(self, project, domain, inputs=None, name=None, notification_overrides=None, label_overrides=None,
+                annotation_overrides=None):
+        """
+        Deprecated. Use launch instead.
```
[the remark this answers is not in the document]
Comment (katrogan) [reply flytekit/common/mixins/launchable.py]: yep, the sdk was already requiring `deprecation` but that wasn't used anywhere so i swapped out the dep in setup.py
